projects/go-lib/src/lib/components/go-table/go-table.component.spec.ts

```diff
@@ -212,4 +217,686 @@ describe('GoTableComponent', () => {
+      component.localTableConfig.totalCount = 26;
+
+      component.setLastPage();
+
+      expect(component.localTableConfig.pageConfig.offset).toBe(25);
+    });
+  });
+
+  describe('isLastPage', () => {
+    let isOnLastPage: boolean;
+
+    beforeEach(() => {
+      component.localTableConfig.totalCount = 26;
+      component.localTableConfig.pageConfig.perPage = 5;
+    });
+
+    it('should return true if on last page', () => {
+      component.localTableConfig.pageConfig.offset = 25;
+
+      isOnLastPage = component.isLastPage();
+
+      expect(isOnLastPage).toBe(true);
+    });
+
+    it('should return false if not on last page', () => {
+      component.localTableConfig.pageConfig.offset = 20;
+
+      isOnLastPage = component.isLastPage();
+
+      expect(isOnLastPage).toBe(false);
+    });
+  });
+
+  describe('nextPage', () => {
+    beforeEach(() => {
+      spyOn(component.tableChange, 'emit');
+    });
+
+    afterEach(() => {
+      expect(component.tableChange.emit).toHaveBeenCalled();
+    });
+
+    it('should go to the next page', () => {
+      component.localTableConfig.totalCount = 25;
+      component.localTableConfig.pageConfig.offset = 5;
+      component.localTableConfig.pageConfig.perPage = 5;
+
+      component.nextPage();
+
+      expect(component.localTableConfig.pageConfig.offset).toBe(10);
+    });
+  });
+
+  describe('toggleSort', () => {
+    beforeEach(() => {
+      spyOn(component.tableChange, 'emit');
+      component.localTableConfig.tableData = fakeTableData;
+      component.localTableConfig.sortable = true;
+      component.localTableConfig.sortConfig = new GoTableSortConfig({
+        column: 'animal'
+      });
+    });
+
+    it('should not make changes if no tableData', () => {
+      component.localTableConfig.tableData = null;
+
+      component.toggleSort(true, 'animal');
+
+      expect(component.tableChange.emit).not.toHaveBeenCalled();
+    });
+
+    it('should not make changes if column is not sortable', () => {
+      component.toggleSort(false, 'animal');
+
+      expect(component.tableChange.emit).not.toHaveBeenCalled();
+    });
+
+    it('should not make changes if table is not sortable', () => {
+      component.localTableConfig.sortable = false;
+
+      component.toggleSort(undefined, 'animal');
+
+      expect(component.tableChange.emit).not.toHaveBeenCalled();
+    });
+
+    it('should make changes if table is sortable', () => {
+      component.toggleSort(undefined, 'animal');
+
+      expect(component.tableChange.emit).toHaveBeenCalled();
+    });
+
+    it('should make changes if column is sortable', () => {
+      component.localTableConfig.sortable = false;
+
+      component.toggleSort(true, 'animal');
+
+      expect(component.tableChange.emit).toHaveBeenCalled();
+    });
+
+    it('should switch sort direction if already sorting on that column', () => {
+      component.toggleSort(undefined, 'animal');
```

Comment:
Should we set the `sortDirection` to `SortDirection.ascending` explicitly before calling `toggleSort` to ensure it get's flipped?

Reply:
In the beforeEach we're setting the sortConfig to a new SortConfig which defaults the sort direction to ascending, but ya we can set it explicitly here.